nf_core/modules.py

```diff
@@ -231,3 +234,81 @@ def has_valid_pipeline(self):
         if not os.path.exists(main_nf) and not os.path.exists(nf_config):
             log.error("Could not find a main.nf or nextfow.config file in: {}".format(self.pipeline_dir))
             return False
+
+
+class ModulesTestHelper(object):
+    def __init__(self, modules_dir=""):
+        self.modules_dir = modules_dir
+
+    # Add custom dumper class to prevent overwriting the global state
+    # This prevents yaml from changing the output order
+    # See https://stackoverflow.com/a/52621703/1497385
+    class CustomDumper(yaml.Dumper):
+        def represent_dict_preserve_order(self, data):
+            return self.represent_dict(data.items())
+
+    CustomDumper.add_representer(dict, CustomDumper.represent_dict_preserve_order)
+
+    def _md5(self, fname):
+        """Generate md5 sum for file"""
+        hash_md5 = hashlib.md5()
+        with open(fname, "rb") as f:
+            for chunk in iter(lambda: f.read(4096), b""):
+                hash_md5.update(chunk)
+        md5sum = hash_md5.hexdigest()
+        return md5sum
+
+    def _get_md5_sums(self, dir, md5_sums=None):
+        """
+        Recursively go through directories and subdirectories
+        and generate tuples of (<file_path>, <md5sum>)
+        returns: list of tuples
+        """
+        if not md5_sums:
+            md5_sums = []
+        elements = glob.glob(dir + "/*")
```

Comment:
Any reason not to use `os.walk`? That would be the usual way to search recursively through a file tree like this..

https://www.tutorialspoint.com/python/os_walk.htm

Reply:
Mainly that I'm used to using glob :) will have a go at walk though. maybe we don't need the recursion then.

Reply:
Oh yes much nicer now!